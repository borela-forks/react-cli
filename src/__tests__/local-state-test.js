// @flow strict

import * as React from "react";
import ReactCLI from "../index";

class TestComponent extends React.Component<{}, { count: number }> {
  state = {
    count: 0
  };
  render() {
    return <div>Counter: {this.state.count}</div>;
  }
}

const componentReference = React.createRef();
let updateCount = 0;
test("components should be able to use local state to manage updates", done => {
  ReactCLI(
    <TestComponent ref={componentReference} />,
    () => {
      if (componentReference.current) {
        componentReference.current.setState({ count: 1 });
      }
    },
    50,
    outputString => {
      updateCount++;
      // this should get called twice
      expect(outputString).toMatchSnapshot();
      if (updateCount === 2) {
        done();
      }
    },
    "+"
  );
});
