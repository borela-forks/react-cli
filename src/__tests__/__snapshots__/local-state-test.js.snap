// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components should be able to use local state to manage updates 1`] = `
"Counter: 0++++++++++++++++++++++++++++++++++++++++
"
`;

exports[`components should be able to use local state to manage updates 2`] = `
"Counter: 1++++++++++++++++++++++++++++++++++++++++
"
`;
